### ISSUE TITLE
Stream not closed?

### ISSUE THREAD

**Issue by manu-silicon** [jbevain/cecil#284]
Just started to use the version from master and it seems that cecil keeps a lock on the assemblies it processes. I mean that I used to be able to delete some assemblies I had generated, but cannot anymore.

I haven't been able to debug this further but wanted to check if some internals have changed that could explain this.

**Comment by SimonCropp** [comment]
@manu-silicon ModuleDefinition is now disposable https://github.com/jbevain/cecil/blob/master/Mono.Cecil/ModuleDefinition.cs#L235  

so r u calling dispose?

**Comment by jbevain** [comment]
@manu-silicon that's intended. We're working on getting a beta of 0.10 out of the door. Here's the relevant entry for the changelog:

### Lower Memory Usage

Cecil used to read all assemblies in memory which had a severe impact on memory when reading large assemblies or assemblies with large resources. Now, by default, a `ModuleDefinition` keeps an open handle on its file and *needs to be disposed*. The default behavior for a `DefaultAssemblyResolver`, which is caching, is to dispose all cached assemblies when the resolver is disposed.

You may override that by setting the `InMemory = true` of the `ReaderParameters`, but that will lead to more memory fragmentation.

Because of this, if you were previously reading and writing an assembly on the same file, you need to change how you read and write your assembly to:

```csharp
// ReaderParameters { ReadWrite = true } is necessary to later write the file
using (var module = ModuleDefinition.ReadModule(file, new ReaderParameters { ReadWrite = true })) 
{
    // Modify the assembly

    module.Write (); // Write to the same file that was used to open the file
}
```

**Comment by manu-silicon** [comment]
@jbevain I managed to find all the placed where I needed to call Dispose(). This was not so easy to migrate the code.

Also even if we used to read/write on the same assembly, we used AssemblyDefinition.Write using a filename as argument (sometime the same file, sometime not). This doesn't work anymore. Maybe adding a new Write method without filename or stream argument in AssemblyDefinition would help.
